test/startservers.py

```diff
@@ -58,13 +64,38 @@ def start():
             'cmd/boulder-va',
             'test/dns-test-srv']:
         try:
-            processes.append(run(prog))
+            processes.append(run(prog, race_detection))
         except Exception as e:
             print(e)
             return False
         if not check():
             # Don't keep building stuff if a server has already died.
             return False
+
+    # Wait until all servers are up before returning to caller. This means
+    # checking each server's debug port until it's available.
+    # seconds.
+    while True:
+        try:
+            # If one of the servers has died, quit immediately.
+            if not check():
+                return False
+            for debug_port in range(8000, 8005):
```

Comment:
If this were only Python 3, you could spin 6 tasks to a thread pool executor and just ask all the futures to complete successfully!

Oh well. :)